src/frontend/hooks/useSettingsValue.ts

```diff
@@ -0,0 +1,7 @@
+import * as React from "react";
+import SettingsContext, { SettingsState } from "../context/SettingsContext";
+
+export default function useSettingsValue(key: keyof SettingsState) {
+  const [state] = React.useContext(SettingsContext);
+  return state[key];
+}
```

Comment:
To improve type inference, we can add a generic here:

```suggestion
export default function useSettingsValue<K extends keyof SettingsState>(
  key: K
): SettingsState[K] {
  const [state] = React.useContext(SettingsContext);
  return state[key];
}
```